### PR TITLE
Documentation update for Content-Type header

### PR DESCRIPTION
**Pull request by csanders-git**
The documentation should be updated to reflect the current RFC 7231 and it's wording as suggested by @mnot in #266. Thanks again!

### CODE REVIEW

File: rules/REQUEST-941-APPLICATION-ATTACK-XSS.conf [hunk 1]
```diff
@@ -128,7 +128,7 @@ SecRule REQUEST_COOKIES|!REQUEST_COOKIES:/__utm/|REQUEST_COOKIES_NAMES|REQUEST_H
 #
 # -=[ XSS Filters - Category 3 ]=-
 #
-SecRule REQUEST_COOKIES|!REQUEST_COOKIES:/__utm/|REQUEST_COOKIES_NAMES|REQUEST_HEADERS:User-Agent|REQUEST_HEADERS:Referer|ARGS_NAMES|ARGS|XML:/* "(?i)[\s\S](?:x(?:link:href|html|mlns)|!ENTITY.*?SYSTEM|data:text\/html|pattern(?=.*?=)|formaction|\@import|base64)[\s\S]" \
+SecRule REQUEST_COOKIES|!REQUEST_COOKIES:/__utm/|REQUEST_COOKIES_NAMES|REQUEST_HEADERS:User-Agent|ARGS_NAMES|ARGS|XML:/* "(?i)[\s\S](?:x(?:link:href|html|mlns)|!ENTITY.*?SYSTEM|data:text\/html|pattern(?=.*?=)|formaction|\@import|base64)[\s\S]" \
```
Comment (dune73) [inline rules/REQUEST-941-APPLICATION-ATTACK-XSS.conf]: These are changes, that are also part of PR #585. Probably remove them here.

File: rules/REQUEST-920-PROTOCOL-ENFORCEMENT.conf [hunk 2]
```diff
@@ -722,15 +722,17 @@ SecRule REQUEST_HEADERS:User-Agent "^$" \
...
-# not empty or contains a 0.  If the Content-Length header contains other data
-# than this means that there is a request body and the RFC states that there
-# MUST be a Content-Type header so that the app knows how to parse the data.
+# -=[ Rule Logic]=-
+# This rule will first check to see if the value of the Content-Length header is
+# non-equal to 0. The chained rule is then checking the existence of the
+# Content-Type header. The RFCs do not state there must be a
+# Content-Type header. However, a request missing a Content-Header is a
+# strong indication of a non-compliant browser.
 #
```
Comment (dune73) [inline rules/REQUEST-920-PROTOCOL-ENFORCEMENT.conf]: Perfect.

File: rules/REQUEST-941-APPLICATION-ATTACK-XSS.conf (hunk 1 above)
Comment (csanders-git) [inline rules/REQUEST-941-APPLICATION-ATTACK-XSS.conf]: I think i just took an updated branch so it wouldn't conflict

Comment (csanders-git) [inline rules/REQUEST-941-APPLICATION-ATTACK-XSS.conf]: will merging cause an issue here @dune73 ?

**Comment by csanders-git** [comment]
whats left on this @dune73, anything?

**Comment by dune73** [comment]
Sorry for the delay. I'm currently looking through 580 and will be right back here afterwards.

File: rules/REQUEST-941-APPLICATION-ATTACK-XSS.conf (hunk 1 above)
Comment (dune73) [inline rules/REQUEST-941-APPLICATION-ATTACK-XSS.conf]: Do not think so. I'll approve.

**Comment by dune73** [comment]
Ready to merge as far as I am concerned.  Sorry for the roadblock with the request for change.

**Comment by csanders-git** [comment]
No problem, we all get busy at various points - i'm a perfect example of that! Thanks for the feedback!

**Pull request merged**